crates/ball_filter/src/hypothesis/moving.rs

```diff
@@ -0,0 +1,71 @@
+use std::time::Duration;
+
+use coordinate_systems::Ground;
+use filtering::kalman_filter::KalmanFilter;
+use linear_algebra::{point, vector, Isometry2, Point2, Vector2};
+use nalgebra::{matrix, Matrix2, Matrix2x4, Matrix4, Matrix4x2};
+use path_serde::{PathDeserialize, PathIntrospect, PathSerialize};
+use serde::{Deserialize, Serialize};
+use types::multivariate_normal_distribution::MultivariateNormalDistribution;
+
+#[derive(Clone, Debug, Serialize, Deserialize, PathSerialize, PathDeserialize, PathIntrospect)]
+pub struct MovingHypothesis(pub MultivariateNormalDistribution<4>);
+
+impl MovingHypothesis {
+    pub fn new(mean: nalgebra::Vector4<f32>, covariance: Matrix4<f32>) -> Self {
+        Self(MultivariateNormalDistribution { mean, covariance })
+    }
+
+    pub fn position(&self) -> Point2<Ground> {
+        point![self.0.mean.x, self.0.mean.y]
+    }
+
+    pub fn velocity(&self) -> Vector2<Ground> {
+        vector![self.0.mean.z, self.0.mean.w]
+    }
+
+    pub fn predict(
+        &mut self,
+        delta_time: Duration,
+        last_to_current_odometry: Isometry2<Ground, Ground>,
+        velocity_decay: f32,
+        process_noise: Matrix4<f32>,
+    ) {
+        let dt = delta_time.as_secs_f32();
+        let constant_velocity_prediction = matrix![
+            1.0, 0.0, dt, 0.0;
+            0.0, 1.0, 0.0, dt;
+            0.0, 0.0, velocity_decay, 0.0;
+            0.0, 0.0, 0.0, velocity_decay;
```

Comment:
Theoretically, yes. But since the velocity update is not linear in time, it is not possible to do it correctly in my understanding. Feel free to propose a solution.

Reply:
hmmm... i may have to sit down with paper and pen. But there might be a proper analytical solution for this. Nevertheless, this might be totally unnecessary to have this correct at this point as `dt` is nearly constant `12ms` anyway...

if you want to keep it this way, feel free to resolve